Add news about sprints (#105)
djangocon/2023.djangocon.eu: _posts/2023-05-12-sprint-attendees.md

---
author: DjangoCon Europe Organisers
category: General
date: 2023-05-12
layout: post
published: true
title: "Sprints: What are they & why should I attend?"
---

> (In the ancient days of FLOSS…) Development sprints are most common around Free/Libre Open Source Software, or FLOSS projects. Most of these projects (like Django and Python) are maintained by a community of volunteers. These volunteers usually work together while spread around the world. Conferences presented rare occasions for these developers to work together in person. And thus, sprints were born! In the context of a given conference or event, you’ll often hear them referred to as “the sprints” or just “sprints”.

Reviewer: ```suggestion
(In the ancient days of FLOSS…) Development sprints are most common around Free/Libre Open Source Software, or FLOSS projects. Most of these projects (like Django and Python) are maintained by a community of volunteers. These volunteers usually work together while spread around the world. Conferences present a rare occasion for these developers to meet in person. And thus, sprints were born! In the context of a given conference or event, you’ll often hear them referred to as “the sprints” or just “sprints”.
```

Over time, these core developers realized sprints also provided them an opportunity to help find new potential contributors. And that’s where YOU come in.

## Why Attend a Sprint at DJEU 2023?
Sprints give you an opportunity to work shoulder-to-shoulder with experts in the community. In addition, there are practical and personal benefits to attending the sprints.

> From a practical perspective, if your organization makes heavy use of Django or a related package, getting involved in its maintenance will give you a better understanding of that package, as well as it’s strengths & limitations. On the personal side, most people won’t usually work on a project that’s as widely used as Django or some of the other related packages. Getting involved in the maintenance of one of these packages can expose you to engineering practices you like wouldn’t see otherwise. This can help you become a better engineer while allowing you to contribute to the community that benefits you and your company.

Reviewer: it's == it is
A check is to expand the contraction: "as well as it is strengths" does not make sense.

```suggestion
From a practical perspective, if your organization makes heavy use of Django or a related package, getting involved in its maintenance will give you a better understanding of that package, as well as its strengths and limitations. On the personal side, most people won’t usually work on a project that’s as widely used as Django or some of the other related packages. Getting involved in the maintenance of one of these packages can expose you to engineering practices you wouldn’t see otherwise. This can help you become a better engineer while allowing you to contribute to the community that benefits you and your company.
```
Not understanding the second clause of "On the personal side". They won't work on these packages so they should get involved in maintaining them?

Author: It was like before it's professional part and then personal part. When we are talking about personal projects, most of the time you don't have many projects to maintain a long period and in Django except a blog for exemple. 
Does it make more sense?

## What should you know if you want to sprint?
Git and Github basics

> Most FLOSS projects these days use Git for version control (the software used for tracking changes to software). This makes it easier for groups of people to work on the same software, together. If you’re new to Git, I recommend the first three chapters of Pro Git, which is freely available to read online or to download. Those chapters will get you started and give you about 80% of the Git you’ll use most often. Chapters five and six, on Distributed Git and Github respectively, are also very helpful if you’re new to working with a group on software.

Reviewer: Could we put a link to the Pro Git book? This one, right? 
https://git-scm.com/book/en/v2

If you’re new to Git or working on software in teams, a basic knowledge of Git fundamentals is important to helping maintainers. Keep in mind that one of the main focuses of contributing to a FLOSS project is to help the people who maintain the software we all use. So, while the sprints aren’t the best place to learn Git, they are a great opportunity to practice or improve your Git skills, especially in situations you may not run into normally.

> In the spirit of helping our maintainers, try to focus on learning to make “good” commits. More specifically, you want to make atomic commits, in which each commit is focused around a specific activity or code change, and clear commit messages, meaning that the changes in a commit should be clear from its commit message. This is important because commits you make will be relied upon by current or future contributors & maintainers. Making “good” commits and messages is both an art and a science, so practice or experience helps. You should also have a GitHub account. Git and GitHub, while related, are two different things, so getting Git setup on your computer won’t give you a GitHub account, just as creating a GitHub account doesn’t mean you’ll have Git installed on your computer.

Reviewer: ```suggestion
In the spirit of helping our maintainers, try to focus on learning how to make “good” commits. More specifically, you want to make atomic commits, in which each commit is focused around a specific activity or code change, with clear commit messages, meaning that the changes in a commit should be clear from its commit message. This is important because the commits you make will be relied upon by current or future contributors and maintainers. Making “good” commits and messages is both an art and a science, so practice and experience helps. You should also have a GitHub account. Git and GitHub, while related, are two different things, so getting Git setup on your computer won’t give you a GitHub account, just as creating a GitHub account doesn’t mean you’ll have Git installed on your computer.
```

## Know yourself
It’s helpful to have an idea of where you are in your development journey. Are you a very experienced developer who’s always wanted to contribute to open source but never has? Are you newer to software development or Django but still interested in contributing? There are multiple ways newer developers can contribute, so don’t let the fear of “not knowing enough” stop you.

> Why do you want to contribute to FLOSS? There are many different reasons, some of which were mentioned above. Are you more focused on giving back to a community that’s already given to you? Maybe you’re looking forward to improving your engineering skills or learning more about a package you and/or your company frequently use or depend on? A little introspection on your motives can help guide your efforts.

Reviewer: ```suggestion
Why do you want to contribute to FLOSS? There are many different reasons, some of which are mentioned above. Are you more focused on giving back to a community that’s already given to you? Maybe you’re looking forward to improving your engineering skills or learning more about a package you and/or your company frequently use or rely on? A little introspection on your motives can help guide your efforts.
```

> Finally, it’s helpful to give some thought to which FLOSS project you want to contribute to. With that in mind, even if there’s not a dedicated sprint happening for that project, you could use the time during the sprints to start looking at making contributions on your own. A one-person sprint is very much “a thing.”

Reviewer: ```suggestion
Finally, it’s helpful to give some thought as to which FLOSS project you want to contribute to. With that in mind, even if there’s not a dedicated sprint happening for that project, you could use the time during the sprints to start looking at making contributions on your own. A one-person sprint is very much “a thing.”
```

---
author: DjangoCon Europe Organisers
category: General

> date: 2023-05-12

Reviewer 2: Not super important, but if you're changing other things can you also update the date of the post

## Questions and Answers
### I’ve never taken part in a sprint before though. So can I join in?

Absolutely. Everyone’s welcome. It’s not an elite club, or for experts only. All of the projects that will be running sprints welcome anyone who wants to help.

### What do I need?

You'll need to know the basics of using Git/GitHub to check out code and make pull requests, to know how to use the issue tracker, and so on. You’ll also need your own computer with a suitable Python environment set up on it - but there'll be people to help with that too.

If you are new to Git, we recommend the first three chapters of [Pro Git](https://git-scm.com/book/en/v2), which is freely available to read online or to download. Those chapters will get you started and give you about 80% of the Git you’ll use most often. Chapters five and six, on Distributed Git and Github respectively, are also very helpful if you’re new to working with a group on software.

### Do I have to stay for a whole sprint?

Not at all. Drop in just for a bit, if you just want to see what it’s like.

> ### Is it needed to good Python programmer?

Reviewer 2: ```suggestion
### Do I need to be a good Python programmer?
```

> You truly don’t need to be very good Python programmer. In fact, you barely need to be a programmer at all, because even the most novice sprinter can make a very valuable contribution by helping improve documentation, for example.

Reviewer 2: ```suggestion
You truly don’t need to be a very good Python programmer. In fact, you barely need to be a programmer at all, because even the most novice sprinter can make a very valuable contribution by helping improve documentation, for example.
```

> ### Do I need to absolutely finish my work during the sprint? If I'm too slow?

Reviewer 2: ```suggestion
### Do I need to finish my work during the sprint? What if I'm too slow?
```

The sprint is to encourage and help new people. You’ll be able to work at your own pace, learning the things you need to learn as you go along, and you’ll be supported by other people. 

> If you can't finish your work during the sprint, you can always continue it after DjangoCon Europe and keep in touch with the people who helped you with it.

Reviewer: ```suggestion
If you can't finish your work during the sprint, you can always continue after DjangoCon Europe and keep in touch with the people who helped you.
```
